feat: snacks notifier support

sonastea opened the pull request.

vuki656 commented on lua/package-info/ui/generic/loading-status.lua:
```diff
@@ -38,7 +43,7 @@ M.new = function(message)
     }
 
-    if nvim_notify then
+    if nvim_notify or snacks_notifier and config.options.notifications then
```
Should this maybe be like this 
```suggestion
    if (nvim_notify or snacks_notifier) and config.options.notifications then
```

vuki656 reviewed (commented):
Could you also check the comments in this issue https://github.com/vuki656/package-info.nvim/issues/164 and see if it's resolved with this PR?

vuki656 commented:
Seams like tests hang for some reason. Probably due to outdated versions. If you can try to update it/fix it if not, turn if off. I don't think these changes are causing it.

sonastea commented:
I updated the ci to use the latest neovim release and modified two tests since they weren't passing, unsure about register_namespace_spec. The notifications config opts should be working correctly now, at least on my end.

sonastea commented on lua/package-info/ui/generic/loading-status.lua:
```diff
@@ -38,11 +51,11 @@ M.new = function(message)
             title = title,
             icon = SPINNERS[1],
-            timeout = false,
+            timeout = 3000,
```
This happens to fix the notifications disappearing using nvim-notify, but the update_spinner doesn't appear to work (works fine with snacks.notifier).
